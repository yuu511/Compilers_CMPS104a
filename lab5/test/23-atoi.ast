ROOT "<<ROOT>>" 0.0.0 {0} 
FUNCTION "" 6.22.1 {0} void function 
TYPE_ID "" 6.22.1 {0} 
VOID "void" 6.22.1 {0} 
IDENT "fail" 6.22.6 {0} 
PARAM "(" 6.22.11 {0} 
TYPE_ID "" 6.22.12 {1} string variable param lval 
STRING "string" 6.22.12 {0} 
IDENT "expr" 6.22.19 {0} 
TYPE_ID "" 6.22.25 {1} string variable param lval 
STRING "string" 6.22.25 {0} 
IDENT "file" 6.22.32 {0} 
TYPE_ID "" 6.22.38 {1} int variable param lval 
INT "int" 6.22.38 {0} 
IDENT "line" 6.22.42 {0} 
FUNCTION "" 6.24.1 {1} void function 
TYPE_ID "" 6.24.1 {0} 
VOID "void" 6.24.1 {0} 
IDENT "putchr" 6.24.6 {0} 
PARAM "(" 6.24.13 {0} 
TYPE_ID "" 6.24.14 {2} int variable param lval 
INT "int" 6.24.14 {0} 
IDENT "chr" 6.24.18 {0} 
FUNCTION "" 6.25.1 {2} void function 
TYPE_ID "" 6.25.1 {0} 
VOID "void" 6.25.1 {0} 
IDENT "putint" 6.25.6 {0} 
PARAM "(" 6.25.13 {0} 
TYPE_ID "" 6.25.14 {3} int variable param lval 
INT "int" 6.25.14 {0} 
IDENT "num" 6.25.18 {0} 
FUNCTION "" 6.26.1 {3} void function 
TYPE_ID "" 6.26.1 {0} 
VOID "void" 6.26.1 {0} 
IDENT "putstr" 6.26.6 {0} 
PARAM "(" 6.26.13 {0} 
TYPE_ID "" 6.26.14 {4} string variable param lval 
STRING "string" 6.26.14 {0} 
IDENT "str" 6.26.21 {0} 
FUNCTION "" 6.28.1 {4} int function 
TYPE_ID "" 6.28.1 {0} 
INT "int" 6.28.1 {0} 
IDENT "getchr" 6.28.5 {0} 
PARAM "(" 6.28.11 {0} 
FUNCTION "" 6.29.1 {5} string function 
TYPE_ID "" 6.29.1 {0} 
STRING "string" 6.29.1 {0} 
IDENT "getstr" 6.29.8 {0} 
PARAM "(" 6.29.14 {0} 
FUNCTION "" 6.30.1 {6} string function 
TYPE_ID "" 6.30.1 {0} 
STRING "string" 6.30.1 {0} 
IDENT "getln" 6.30.8 {0} 
PARAM "(" 6.30.13 {0} 
FUNCTION "" 7.5.1 {7} int function 
TYPE_ID "" 7.5.1 {0} 
INT "int" 7.5.1 {0} 
IDENT "atoi" 7.5.5 {0} 
PARAM "(" 7.5.10 {0} 
TYPE_ID "" 7.5.11 {8} string variable param lval 
STRING "string" 7.5.11 {0} 
IDENT "str" 7.5.18 {0} 
BLOCK "{" 7.5.23 {8} 
TYPE_ID "" 7.6.4 {8} int variable local lval vreg (7.6.4)
INT "int" 7.6.4 {0} 
IDENT "neg" 7.6.8 {0} 
INTCON "0" 7.6.14 {8} int const 
TYPE_ID "" 7.7.4 {8} int variable local lval vreg (7.7.4)
INT "int" 7.7.4 {0} 
IDENT "num" 7.7.8 {0} 
INTCON "0" 7.7.14 {8} int const 
TYPE_ID "" 7.8.4 {8} int variable local lval vreg (7.8.4)
INT "int" 7.8.4 {0} 
IDENT "digit" 7.8.8 {0} 
INTCON "0" 7.8.16 {8} int const 
BLOCK "{" 7.9.4 {8} 
IF "if" 7.9.5 {8} 
NOT "not" 7.9.9 {8} int vreg 
NE "!=" 7.9.18 {8} int vreg 
IDENT "str" 7.9.14 {8} string variable param lval (7.5.11)
NULLPTR "nullptr" 7.9.21 {8} nullptr_t const 
CALL "(" 7.9.36 {0} void function (6.22.1)
IDENT "fail" 7.9.31 {0} 
STRINGCON ""str != nullptr"" 7.9.37 {8} string const 
STRINGCON ""23-atoi.oc"" 7.9.55 {8} string const 
INTCON "9" 7.9.69 {8} int const 
';' ";" 7.9.73 {0} 
IF "if" 7.10.4 {8} 
NE "!=" 7.10.15 {8} int vreg 
INDEX "[" 7.10.11 {8} int lval vaddr 
IDENT "str" 7.10.8 {8} string variable param lval (7.5.11)
INTCON "0" 7.10.12 {8} int const 
CHARCON "'\0'" 7.10.18 {8} int const 
BLOCK "{" 7.10.24 {8} 
IF "if" 7.11.7 {8} 
EQ "==" 7.11.18 {8} int vreg 
INDEX "[" 7.11.14 {8} int lval vaddr 
IDENT "str" 7.11.11 {8} string variable param lval (7.5.11)
INTCON "0" 7.11.15 {8} int const 
CHARCON "'-'" 7.11.21 {8} int const 
BLOCK "{" 7.11.26 {8} 
'=' "=" 7.12.16 {8} int variable local lval vreg 
IDENT "digit" 7.12.10 {8} int variable local lval (7.8.4)
'+' "+" 7.12.24 {8} int vreg 
IDENT "digit" 7.12.18 {8} int variable local lval (7.8.4)
INTCON "1" 7.12.26 {8} int const 
'=' "=" 7.13.14 {8} int variable local lval vreg 
IDENT "neg" 7.13.10 {8} int variable local lval (7.6.4)
INTCON "1" 7.13.16 {8} int const 
TYPE_ID "" 7.15.7 {8} int variable local lval vreg (7.15.7)
INT "int" 7.15.7 {0} 
IDENT "contin" 7.15.11 {0} 
INTCON "1" 7.15.20 {8} int const 
WHILE "while" 7.16.7 {8} 
IDENT "contin" 7.16.14 {8} int variable local lval (7.15.7)
BLOCK "{" 7.16.22 {8} 
IF "if" 7.17.10 {8} 
EQ "==" 7.17.25 {8} int vreg 
INDEX "[" 7.17.17 {8} int lval vaddr 
IDENT "str" 7.17.14 {8} string variable param lval (7.5.11)
IDENT "digit" 7.17.18 {8} int variable local lval (7.8.4)
CHARCON "'\0'" 7.17.28 {8} int const 
BLOCK "{" 7.17.34 {8} 
'=' "=" 7.18.20 {8} int variable local lval vreg 
IDENT "contin" 7.18.13 {8} int variable local lval (7.15.7)
INTCON "0" 7.18.22 {8} int const 
BLOCK "{" 7.19.16 {8} 
TYPE_ID "" 7.20.13 {8} int variable local lval vaddr (7.20.13)
INT "int" 7.20.13 {0} 
IDENT "chr" 7.20.17 {0} 
INDEX "[" 7.20.26 {8} int lval vaddr 
IDENT "str" 7.20.23 {8} string variable param lval (7.5.11)
IDENT "digit" 7.20.27 {8} int variable local lval (7.8.4)
'=' "=" 7.21.19 {8} int variable local lval vreg 
IDENT "digit" 7.21.13 {8} int variable local lval (7.8.4)
'+' "+" 7.21.27 {8} int vreg 
IDENT "digit" 7.21.21 {8} int variable local lval (7.8.4)
INTCON "1" 7.21.29 {8} int const 
IF "if" 7.22.13 {8} 
LT "<" 7.22.21 {8} int vreg 
IDENT "chr" 7.22.17 {8} int variable local lval (7.20.13)
CHARCON "'0'" 7.22.23 {8} int const 
'=' "=" 7.22.35 {8} int variable local lval vreg 
IDENT "contin" 7.22.28 {8} int variable local lval (7.15.7)
INTCON "0" 7.22.37 {8} int const 
IF "if" 7.23.18 {8} 
GT ">" 7.23.26 {8} int vreg 
IDENT "chr" 7.23.22 {8} int variable local lval (7.20.13)
CHARCON "'9'" 7.23.28 {8} int const 
'=' "=" 7.23.40 {8} int variable local lval vreg 
IDENT "contin" 7.23.33 {8} int variable local lval (7.15.7)
INTCON "0" 7.23.42 {8} int const 
'=' "=" 7.24.22 {8} int variable local lval vreg 
IDENT "num" 7.24.18 {8} int variable local lval (7.7.4)
'-' "-" 7.24.39 {8} int vreg 
'+' "+" 7.24.33 {8} int vreg 
'*' "*" 7.24.28 {8} int vreg 
IDENT "num" 7.24.24 {8} int variable local lval (7.7.4)
INTCON "10" 7.24.30 {8} int const 
IDENT "chr" 7.24.35 {8} int variable local lval (7.20.13)
CHARCON "'0'" 7.24.41 {8} int const 
IF "if" 7.27.7 {8} 
IDENT "neg" 7.27.11 {8} int variable local lval (7.6.4)
'=' "=" 7.27.20 {8} int variable local lval vreg 
IDENT "num" 7.27.16 {8} int variable local lval (7.7.4)
'-' "-" 7.27.22 {8} int variable local lval vreg 
IDENT "num" 7.27.24 {8} int variable local lval (7.7.4)
RETURN "return" 7.29.4 {8} 
IDENT "num" 7.29.11 {8} int variable local lval (7.7.4)
FUNCTION "" 7.32.1 {8} int function 
TYPE_ID "" 7.32.1 {0} 
INT "int" 7.32.1 {0} 
IDENT "main" 7.32.5 {0} 
PARAM "(" 7.32.10 {0} 
TYPE_ID "" 7.32.11 {9} int variable param lval 
INT "int" 7.32.11 {0} 
IDENT "argc" 7.32.15 {0} 
TYPE_ID "" 7.32.21 {9} string array variable param lval 
ARRAY "array" 7.32.21 {0} 
STRING "string" 7.32.27 {0} 
IDENT "argv" 7.32.35 {0} 
BLOCK "{" 7.32.41 {9} 
TYPE_ID "" 7.33.4 {9} int variable local lval vreg (7.33.4)
INT "int" 7.33.4 {0} 
IDENT "argi" 7.33.8 {0} 
INTCON "1" 7.33.15 {9} int const 
TYPE_ID "" 7.34.4 {9} string variable local lval vreg (7.34.4)
STRING "string" 7.34.4 {0} 
IDENT "arg" 7.34.11 {0} 
NULLPTR "nullptr" 7.34.17 {9} nullptr_t const 
WHILE "while" 7.35.4 {9} 
LT "<" 7.35.16 {9} int vreg 
IDENT "argi" 7.35.11 {9} int variable local lval (7.33.4)
IDENT "argc" 7.35.18 {9} int variable param lval (7.32.11)
BLOCK "{" 7.35.24 {9} 
'=' "=" 7.36.11 {9} string variable local lval vaddr 
IDENT "arg" 7.36.7 {9} string variable local lval (7.34.4)
INDEX "[" 7.36.17 {9} string lval vaddr 
IDENT "argv" 7.36.13 {9} string array variable param lval (7.32.21)
IDENT "argi" 7.36.18 {9} int variable local lval (7.33.4)
CALL "(" 7.37.14 {3} void function (6.26.1)
IDENT "putstr" 7.37.7 {0} 
IDENT "arg" 7.37.15 {9} string variable local lval (7.34.4)
CALL "(" 7.38.14 {3} void function (6.26.1)
IDENT "putstr" 7.38.7 {0} 
STRINGCON "" = "" 7.38.15 {9} string const 
CALL "(" 7.39.14 {2} void function (6.25.1)
IDENT "putint" 7.39.7 {0} 
CALL "(" 7.39.20 {7} int function (7.5.1)
IDENT "atoi" 7.39.15 {0} 
IDENT "arg" 7.39.21 {9} string variable local lval (7.34.4)
CALL "(" 7.40.14 {1} void function (6.24.1)
IDENT "putchr" 7.40.7 {0} 
CHARCON "'\n'" 7.40.15 {9} int const 
'=' "=" 7.41.12 {9} int variable local lval vreg 
IDENT "argi" 7.41.7 {9} int variable local lval (7.33.4)
'+' "+" 7.41.19 {9} int vreg 
IDENT "argi" 7.41.14 {9} int variable local lval (7.33.4)
INTCON "1" 7.41.21 {9} int const 
RETURN "return" 7.43.4 {9} 
INTCON "0" 7.43.11 {9} int const 
