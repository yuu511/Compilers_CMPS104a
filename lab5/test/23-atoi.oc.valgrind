==6581== Memcheck, a memory error detector
==6581== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==6581== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==6581== Command: ../oc 23-atoi.oc
==6581== 
==6581== 
==6581== HEAP SUMMARY:
==6581==     in use at exit: 0 bytes in 0 blocks
==6581==   total heap usage: 1,173 allocs, 1,173 frees, 190,101 bytes allocated
==6581== 
==6581== All heap blocks were freed -- no leaks are possible
==6581== 
==6581== For counts of detected and suppressed errors, rerun with: -v
==6581== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
