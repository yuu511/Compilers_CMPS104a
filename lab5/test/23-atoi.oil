atoi:      .function int 
.param ptr str
.local int neg
.local int num
.local int digit
.local int contin
.local int chr
return
.end
main:      .function int 
.param int argc
.param ptr argv
.local int argi
.local ptr arg
return
.end
