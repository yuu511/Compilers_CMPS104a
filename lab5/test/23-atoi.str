string_set[   2]:    2594088158685378223 0x56551e8d0d98->"10"
                   12597418120160561823 0x56551e8ca7b8->"int"
string_set[   3]:   11752130776608228127 0x56551e8cde78->"'\0'"
string_set[   5]:    1370730858159036685 0x56551e8ce718->"+"
string_set[   8]:   13497578857207247412 0x56551e8ca238->"string"
string_set[   9]:   10319632053171904911 0x56551e8cecb8->"while"
string_set[  10]:    9201800968376702500 0x56551e8cbb68->"getchr"
string_set[  12]:   14643508047410943861 0x56551e8d1738->"argv"
                    1902468552727064667 0x56551e8ccfd8->"digit"
string_set[  14]:   15816506932648033945 0x56551e8ca178->"("
                    1396094175042499165 0x56551e8cfe48->"<"
string_set[  15]:   12868445110721718657 0x56551e8ca958->")"
string_set[  17]:    6142509188972423790 0x56551e8ca418->""
string_set[  20]:    1583576014053361350 0x56551e8cb848->"str"
string_set[  21]:    9317282556992397945 0x56551e8cd278->"if"
string_set[  22]:   10159970873491820195 0x56551e8ce7d8->"1"
string_set[  26]:   12291392672316406109 0x56551e8ca878->"line"
                    9611428142580219387 0x56551e8ca0b8->"fail"
                   11597697714117577063 0x56551e8ccb48->"="
string_set[  28]:   18413295556186398326 0x56551e8cb158->"putint"
string_set[  29]:   11911796782241914524 0x56551e8c8fe8->"void"
string_set[  33]:    6814028903552918164 0x56551e8d1558->"array"
string_set[  35]:   11868120863286193964 0x56551e8d1358->"main"
string_set[  41]:   11038708016986559757 0x56551e8d07d8->"'9'"
                   12051031859636400188 0x56551e8cdd78->"]"
string_set[  42]:   13686721134187162920 0x56551e8cdb28->"}"
string_set[  43]:   14849783028996024063 0x56551e8cd3b8->"not"
string_set[  45]:    1145006563990091304 0x56551e8cca88->"neg"
string_set[  49]:   12679825168387489523 0x56551e8cb318->"num"
string_set[  50]:      21606623943142546 0x56551e8cc7c8->"{"
string_set[  52]:    5344662657899890615 0x56551e8caad8->";"
string_set[  54]:    7296690267450279874 0x56551e8cadf8->"chr"
string_set[  55]:    2297668033614959926 0x56551e8ccc08->"0"
string_set[  56]:    3701972582333163920 0x56551e8d1398->"argc"
string_set[  59]:    6192799629375248351 0x56551e8d2b88->"'\n'"
string_set[  60]:    1873634190020599143 0x56551e8cf538->"else"
string_set[  63]:   14526079891305879776 0x56551e8d0718->">"
                    6378628904606585349 0x56551e8cc468->"atoi"
string_set[  64]:   16579916199511856337 0x56551e8cc168->"getln"
string_set[  67]:    4706206694200543916 0x56551e8ca4f8->","
                   12808232827189627082 0x56551e8cd638->"nullptr"
string_set[  69]:    5266873771696973564 0x56551e8d1938->"argi"
string_set[  72]:    7869624569597532036 0x56551e8cd978->""23-atoi.oc""
string_set[  73]:    9386170080689500517 0x56551e8ca2f8->"expr"
string_set[  76]:   11166701219533395393 0x56551e8cbe68->"getstr"
string_set[  78]:    7761470112706414413 0x56551e8d1bc8->"arg"
string_set[  80]:    7211130878751977441 0x56551e8cac38->"putchr"
                    3729804957429652673 0x56551e8c3dd8->"<<ROOT>>"
                    2617056156435215158 0x56551e8cd578->"!="
string_set[  83]:   13581035385482130682 0x56551e8cdc38->"["
string_set[  84]:    5720712447182791414 0x56551e8cff08->"'0'"
string_set[  85]:    7372855250431704371 0x56551e8d0f78->"-"
                    6006759838892838694 0x56551e8ce3b8->"'-'"
                    4017852018407011111 0x56551e8ca638->"file"
string_set[  86]:   17605730733847049629 0x56551e8ceab8->"contin"
string_set[  87]:   14635384904416252381 0x56551e8d1268->"return"
string_set[  89]:    6637313742931709005 0x56551e8d0cd8->"*"
                   13348923631108115385 0x56551e8cb688->"putstr"
                   13599092719175287697 0x56551e8cd7f8->""str != nullptr""
                    6623666755989985924 0x56551e8cda68->"9"
string_set[  91]:    5090247647583737318 0x56551e8d26d8->"" = ""
string_set[  95]:   14736352883882276958 0x56551e8ce2f8->"=="
load_factor = 0.639
bucket_count = 97
max_bucket_size = 4
