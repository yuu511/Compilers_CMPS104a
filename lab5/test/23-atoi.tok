#  22 "oclib.h"
    6  22.001 258 TOK_VOID      void
    6  22.006 278 TOK_IDENT     fail
    6  22.011  40 '('           (
    6  22.012 260 TOK_STRING    string
    6  22.019 278 TOK_IDENT     expr
    6  22.023  44 ','           ,
    6  22.025 260 TOK_STRING    string
    6  22.032 278 TOK_IDENT     file
    6  22.036  44 ','           ,
    6  22.038 259 TOK_INT       int
    6  22.042 278 TOK_IDENT     line
    6  22.046  41 ')'           )
    6  22.047  59 ';'           ;
    6  24.001 258 TOK_VOID      void
    6  24.006 278 TOK_IDENT     putchr
    6  24.013  40 '('           (
    6  24.014 259 TOK_INT       int
    6  24.018 278 TOK_IDENT     chr
    6  24.021  41 ')'           )
    6  24.022  59 ';'           ;
    6  25.001 258 TOK_VOID      void
    6  25.006 278 TOK_IDENT     putint
    6  25.013  40 '('           (
    6  25.014 259 TOK_INT       int
    6  25.018 278 TOK_IDENT     num
    6  25.021  41 ')'           )
    6  25.022  59 ';'           ;
    6  26.001 258 TOK_VOID      void
    6  26.006 278 TOK_IDENT     putstr
    6  26.013  40 '('           (
    6  26.014 260 TOK_STRING    string
    6  26.021 278 TOK_IDENT     str
    6  26.024  41 ')'           )
    6  26.025  59 ';'           ;
    6  28.001 259 TOK_INT       int
    6  28.005 278 TOK_IDENT     getchr
    6  28.011  40 '('           (
    6  28.012  41 ')'           )
    6  28.013  59 ';'           ;
    6  29.001 260 TOK_STRING    string
    6  29.008 278 TOK_IDENT     getstr
    6  29.014  40 '('           (
    6  29.015  41 ')'           )
    6  29.016  59 ';'           ;
    6  30.001 260 TOK_STRING    string
    6  30.008 278 TOK_IDENT     getln
    6  30.013  40 '('           (
    6  30.014  41 ')'           )
    6  30.015  59 ';'           ;
#   5 "23-atoi.oc"
    7   5.001 259 TOK_INT       int
    7   5.005 278 TOK_IDENT     atoi
    7   5.010  40 '('           (
    7   5.011 260 TOK_STRING    string
    7   5.018 278 TOK_IDENT     str
    7   5.021  41 ')'           )
    7   5.023 123 '{'           {
    7   6.004 259 TOK_INT       int
    7   6.008 278 TOK_IDENT     neg
    7   6.012  61 '='           =
    7   6.014 279 TOK_INTCON    0
    7   6.015  59 ';'           ;
    7   7.004 259 TOK_INT       int
    7   7.008 278 TOK_IDENT     num
    7   7.012  61 '='           =
    7   7.014 279 TOK_INTCON    0
    7   7.015  59 ';'           ;
    7   8.004 259 TOK_INT       int
    7   8.008 278 TOK_IDENT     digit
    7   8.014  61 '='           =
    7   8.016 279 TOK_INTCON    0
    7   8.017  59 ';'           ;
    7   9.004 123 '{'           {
    7   9.005 261 TOK_IF        if
    7   9.008  40 '('           (
    7   9.009 277 TOK_NOT       not
    7   9.013  40 '('           (
    7   9.014 278 TOK_IDENT     str
    7   9.018 272 TOK_NE        !=
    7   9.021 266 TOK_NULLPTR   nullptr
    7   9.028  41 ')'           )
    7   9.029  41 ')'           )
    7   9.031 278 TOK_IDENT     fail
    7   9.036  40 '('           (
    7   9.037 281 TOK_STRINGCON "str != nullptr"
    7   9.053  44 ','           ,
    7   9.055 281 TOK_STRINGCON "23-atoi.oc"
    7   9.067  44 ','           ,
    7   9.069 279 TOK_INTCON    9
    7   9.070  41 ')'           )
    7   9.071  59 ';'           ;
    7   9.072 125 '}'           }
    7   9.073  59 ';'           ;
    7  10.004 261 TOK_IF        if
    7  10.007  40 '('           (
    7  10.008 278 TOK_IDENT     str
    7  10.011  91 '['           [
    7  10.012 279 TOK_INTCON    0
    7  10.013  93 ']'           ]
    7  10.015 272 TOK_NE        !=
    7  10.018 280 TOK_CHARCON   '\0'
    7  10.022  41 ')'           )
    7  10.024 123 '{'           {
    7  11.007 261 TOK_IF        if
    7  11.010  40 '('           (
    7  11.011 278 TOK_IDENT     str
    7  11.014  91 '['           [
    7  11.015 279 TOK_INTCON    0
    7  11.016  93 ']'           ]
    7  11.018 271 TOK_EQ        ==
    7  11.021 280 TOK_CHARCON   '-'
    7  11.024  41 ')'           )
    7  11.026 123 '{'           {
    7  12.010 278 TOK_IDENT     digit
    7  12.016  61 '='           =
    7  12.018 278 TOK_IDENT     digit
    7  12.024  43 '+'           +
    7  12.026 279 TOK_INTCON    1
    7  12.027  59 ';'           ;
    7  13.010 278 TOK_IDENT     neg
    7  13.014  61 '='           =
    7  13.016 279 TOK_INTCON    1
    7  13.017  59 ';'           ;
    7  14.007 125 '}'           }
    7  15.007 259 TOK_INT       int
    7  15.011 278 TOK_IDENT     contin
    7  15.018  61 '='           =
    7  15.020 279 TOK_INTCON    1
    7  15.021  59 ';'           ;
    7  16.007 263 TOK_WHILE     while
    7  16.013  40 '('           (
    7  16.014 278 TOK_IDENT     contin
    7  16.020  41 ')'           )
    7  16.022 123 '{'           {
    7  17.010 261 TOK_IF        if
    7  17.013  40 '('           (
    7  17.014 278 TOK_IDENT     str
    7  17.017  91 '['           [
    7  17.018 278 TOK_IDENT     digit
    7  17.023  93 ']'           ]
    7  17.025 271 TOK_EQ        ==
    7  17.028 280 TOK_CHARCON   '\0'
    7  17.032  41 ')'           )
    7  17.034 123 '{'           {
    7  18.013 278 TOK_IDENT     contin
    7  18.020  61 '='           =
    7  18.022 279 TOK_INTCON    0
    7  18.023  59 ';'           ;
    7  19.010 125 '}'           }
    7  19.011 262 TOK_ELSE      else
    7  19.016 123 '{'           {
    7  20.013 259 TOK_INT       int
    7  20.017 278 TOK_IDENT     chr
    7  20.021  61 '='           =
    7  20.023 278 TOK_IDENT     str
    7  20.026  91 '['           [
    7  20.027 278 TOK_IDENT     digit
    7  20.032  93 ']'           ]
    7  20.033  59 ';'           ;
    7  21.013 278 TOK_IDENT     digit
    7  21.019  61 '='           =
    7  21.021 278 TOK_IDENT     digit
    7  21.027  43 '+'           +
    7  21.029 279 TOK_INTCON    1
    7  21.030  59 ';'           ;
    7  22.013 261 TOK_IF        if
    7  22.016  40 '('           (
    7  22.017 278 TOK_IDENT     chr
    7  22.021  60 '<'           <
    7  22.023 280 TOK_CHARCON   '0'
    7  22.026  41 ')'           )
    7  22.028 278 TOK_IDENT     contin
    7  22.035  61 '='           =
    7  22.037 279 TOK_INTCON    0
    7  22.038  59 ';'           ;
    7  23.013 262 TOK_ELSE      else
    7  23.018 261 TOK_IF        if
    7  23.021  40 '('           (
    7  23.022 278 TOK_IDENT     chr
    7  23.026  62 '>'           >
    7  23.028 280 TOK_CHARCON   '9'
    7  23.031  41 ')'           )
    7  23.033 278 TOK_IDENT     contin
    7  23.040  61 '='           =
    7  23.042 279 TOK_INTCON    0
    7  23.043  59 ';'           ;
    7  24.013 262 TOK_ELSE      else
    7  24.018 278 TOK_IDENT     num
    7  24.022  61 '='           =
    7  24.024 278 TOK_IDENT     num
    7  24.028  42 '*'           *
    7  24.030 279 TOK_INTCON    10
    7  24.033  43 '+'           +
    7  24.035 278 TOK_IDENT     chr
    7  24.039  45 '-'           -
    7  24.041 280 TOK_CHARCON   '0'
    7  24.044  59 ';'           ;
    7  25.010 125 '}'           }
    7  26.007 125 '}'           }
    7  27.007 261 TOK_IF        if
    7  27.010  40 '('           (
    7  27.011 278 TOK_IDENT     neg
    7  27.014  41 ')'           )
    7  27.016 278 TOK_IDENT     num
    7  27.020  61 '='           =
    7  27.022  45 '-'           -
    7  27.024 278 TOK_IDENT     num
    7  27.027  59 ';'           ;
    7  28.004 125 '}'           }
    7  29.004 264 TOK_RETURN    return
    7  29.011 278 TOK_IDENT     num
    7  29.014  59 ';'           ;
    7  30.001 125 '}'           }
    7  32.001 259 TOK_INT       int
    7  32.005 278 TOK_IDENT     main
    7  32.010  40 '('           (
    7  32.011 259 TOK_INT       int
    7  32.015 278 TOK_IDENT     argc
    7  32.019  44 ','           ,
    7  32.021 267 TOK_ARRAY     array
    7  32.026  60 '<'           <
    7  32.027 260 TOK_STRING    string
    7  32.033  62 '>'           >
    7  32.035 278 TOK_IDENT     argv
    7  32.039  41 ')'           )
    7  32.041 123 '{'           {
    7  33.004 259 TOK_INT       int
    7  33.008 278 TOK_IDENT     argi
    7  33.013  61 '='           =
    7  33.015 279 TOK_INTCON    1
    7  33.016  59 ';'           ;
    7  34.004 260 TOK_STRI