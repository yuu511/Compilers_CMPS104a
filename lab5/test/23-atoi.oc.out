CONDITIONAL IF STATEMENT SYMBOL TOK_NOT 
CONDITIONAL IF STATEMENT SYMBOL TOK_NE 
CONDITIONAL IF STATEMENT SYMBOL TOK_EQ 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_IDENT 
CONDITIONAL IF STATEMENT SYMBOL TOK_EQ 
CONDITIONAL IF STATEMENT SYMBOL TOK_LT 
CONDITIONAL IF STATEMENT SYMBOL TOK_GT 
CONDITIONAL IF STATEMENT SYMBOL TOK_IDENT 
CONDITIONAL WHILE STATEMENT SYMBOL TOK_LT 
